src/shared/listener-collection.js

```diff
@@ -11,25 +11,42 @@
  */
 export class ListenerCollection {
   constructor() {
-    /** @type {Listener[]} */
-    this._listeners = [];
+    /** @type {Map<Symbol, Listener>} */
+    this._listeners = new Map();
   }
 
   /**
+   * Adds a listener and returns a listenerId
+   *
    * @param {Listener['eventTarget']} eventTarget
    * @param {Listener['eventType']} eventType
    * @param {Listener['listener']} listener
    * @param {AddEventListenerOptions} [options]
    */
   add(eventTarget, eventType, listener, options) {
     eventTarget.addEventListener(eventType, listener, options);
-    this._listeners.push({ eventTarget, eventType, listener });
+    const symbol = Symbol();
+    this._listeners.set(symbol, { eventTarget, eventType, listener });
+    return symbol;
```

Comment:
Using a symbol works. I guess you could also have made `this._listeners` a set which stores the `{ eventTarget, eventType, listener }` objects and return the object reference as a key.